src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -801,7 +797,9 @@ static PyObject* char2pylist(char* buf, int np, int* cnt, int* displ) {
             Py_INCREF(Py_None);  // 'Fatal Python error: deallocating None' eventually
             PyList_SetItem(plist, i, Py_None);
         } else {
-            PyObject* p = unpickle(buf + displ[i], cnt[i]);
+            nb::object po = unpickle(buf + displ[i], cnt[i]);
+            po.inc_ref();
+            PyObject* p = po.ptr();
             PyList_SetItem(plist, i, p);
```

Comment:
Can you please check if `object` supports `release()`? I'd prefer the following:
```suggestion
            nb::object po = unpickle(buf + displ[i], cnt[i]);
            PyList_SetItem(plist, i, po.release().ptr());
```

Reply:
Thanks for this `release()`!